mmengine/config/new_config.py

```diff
@@ -0,0 +1,459 @@
+        object."""
+        res = self._cfg_dict._to_lazy_dict()
+
+        if keep_imported:
+            return res
+        else:
+            return dict(filter(filter_imports, res.items()))
+
+    def to_dict(self, keep_imported: bool = False):
+        """Convert all data in the config to a builtin ``dict``.
+
+        Args:
+            keep_imported (bool): Whether to keep the imported field.
+                Defaults to False
+
+        If you import third-party objects in the config file, all imported
+        objects will be converted to a string like ``torch.optim.SGD``
+        """
+        _cfg_dict = self._to_lazy_dict(keep_imported=keep_imported)
+
+        def lazy2string(cfg_dict):
+            if isinstance(cfg_dict, dict):
+                return type(cfg_dict)(
+                    {k: lazy2string(v)
+                     for k, v in cfg_dict.items()})
+            elif isinstance(cfg_dict, (tuple, list)):
+                return type(cfg_dict)(lazy2string(v) for v in cfg_dict)
+            else:
+                dump_str = dump_extra_type(cfg_dict)
+                return dump_str if dump_str is not None else cfg_dict
+
+        return lazy2string(_cfg_dict)
+
+
+class BaseImportContext():
+
+    def __enter__(self):
+        # Disable enabled lazy loader during parsing base
+        self.lazy_importers = []
+        for p in sys.meta_path:
+            if isinstance(p, LazyImportContext) and p.enable:
+                self.lazy_importers.append(p)
+                p.enable = False
+
+        old_import = builtins.__import__
+
+        def new_import(name, globals=None, locals=None, fromlist=(), level=0):
+            cur_file = None
+
+            # Try to import the base config source file
+            if level != 0 and globals is not None:
+                # For relative import path
+                if '__file__' in globals:
+                    loc = Path(globals['__file__']).parent
+                else:
+                    loc = Path(os.getcwd())
+                cur_file = self.find_relative_file(loc, name, level - 1)
+                if not cur_file.exists():
+                    raise ImportError(
+                        f'Cannot find the base config "{name}" from '
+                        f'{loc}: {cur_file} does not exist.')
+            elif level == 0:
+                # For absolute import path
+                pkg, _, mod = name.partition('.')
+                pkg = PathFinder.find_spec(pkg)
+                if mod and pkg.submodule_search_locations:
+                    loc = Path(pkg.submodule_search_locations[0])
+                    cur_file = self.find_relative_file(loc, mod)
+                    if not cur_file.exists():
+                        raise ImportError(
+                            f'Cannot find the base config "{name}": '
+                            f'{cur_file} does not exist.')
+
+            # Recover the original import during handle the base config file.
```

Comment:
```suggestion
            # Recover the lazy import mode when parsing base config file

Reply:
The context doesn't depend on lazy import.